File: sm06/models/animals/base/Animal.java
package sm06.models.animals.base;

/**
 * Animal is ab abstract base class to inherit
 */
public abstract class Animal {
    private double height;
    private double weight;
    private String eyeColor;

    /**
     * It is a constructor for Animal class
     * @param height    The height of animal
     * @param weight    The weight of animal
     * @param eyeColor  The eye color of animal
     */
    protected Animal(double height, double weight, String eyeColor) {
        this.height = height;
        this.weight = weight;
        this.eyeColor = eyeColor;
    }

    /**
     * Returns the animal height
     * @return this.height
     */
    public double getHeight() {
        return this.height;
    }

    /**
     * Returns the animal height
     * @return this.weight
     */
    public double getWeight() {
        return this.weight;
    }

    /**
     * Returns the animal eye color
     * @return this.eyeColor
     */
    public String getEyeColor() {
        return this.eyeColor;
    }

    /**
     * Makes sound of animal
     */
    public abstract String sound();

    @Override
    public String toString() {
        return String.format("Height: %.1f, Height: %.1f, Eye color: %s", this.getHeight(), this.getWeight(), this.getEyeColor());
    }
}

File: hw07/internal/models/animal.go
package models

import "fmt"

// Animal base structure
type Animal struct {
	height   float64
	weight   float64
	eyeColor string
}

// Creates new Animal instance
func newAnimal(height float64, weight float64, eyeColor string) (*Animal, error) {
	a := &Animal{
		height:   height,
		weight:   weight,
		eyeColor: eyeColor,
	}

	// Handles potential restrictions of any parameter to return error when new Animal cannot be instantiated
	return a, nil
}

// Returns the animal height
func (a *Animal) Height() float64 {
	return a.height
}

// Returns the animal weight
func (a *Animal) Weight() float64 {
	return a.weight
}

// Returns the animal eyeColor
func (a *Animal) EyeColor() string {
	return a.eyeColor
}

// Returns the string representation of Animal instance
func (a *Animal) String() string {
	return fmt.Sprintf("Height: %.1f, Height: %.1f, Eye color: %s", a.height, a.weight, a.eyeColor)
}
